processor/src/chiplets/kernel_rom/mod.rs

```diff
@@ -0,0 +1,143 @@
+use super::{BTreeMap, Digest, ExecutionError, Felt, Kernel, TraceFragment, Word, ONE, ZERO};
+use vm_core::chiplets::kernel_rom::TRACE_WIDTH;
+
+#[cfg(test)]
+mod tests;
+
+// KERNEL ROM
+// ================================================================================================
+
+/// Kernel ROM chiplet for the VM.
+///
+/// This component is responsible for validating that kernel calls requested by the executing
+/// program are made against procedures which are contained within the specified kernel. It also
+/// tacks all calls to kernel procedures and this info is used to construct an execution trace of
+/// all kernel accesses.
+///
+/// # Execution trace
+/// The layout of the execution trace of kernel procedure accesses is shown below:
+///
+///   s0   idx   h0   h1   h2   h3
+/// ├────┴─────┴────┴────┴────┴────┤
+///
+/// In the above, the meaning of columns is as follows:
+/// - `s0` is a selector column which indicates whether a procedure in a given row should count
+///   toward kernel access. ONE indicates that a procedure should be counted as a single access,
+///   and ZERO indicates that it shouldn't.
+/// - `idx` is a procedure index in the kernel. Values in this column start at ZERO and are
+///   incremented by ONE for every new procedure. Said another way, if `idx` does not change,
+///   values in `h0` - `h3` must remain the same, but when `idx` is incremented values in `h0` -
+///   `h3` can change.
+/// - `h0` - `h3` columns contain roots of procedures in a given kernel. Together with `idx`
+///   column, these form tuples (index, procedure root) for all procedures in the kernel.
+pub struct KernelRom {
+    access_map: BTreeMap<[u8; 32], ProcAccessInfo>,
```

Comment:
nit: should we consider creating a named type instead of `[u8; 32]`? This improves not only code readability as it will be unmistakable about what is being mapped, but also type safety: a `ProcAccessInfo` can only be fetched from a properly calculated hash

Reply:
I added a type alias to make it a bit more clear. We actually already have a special type for hashes (`Digest`). However,  this type does not implement `Ord` and thus we convert `Digest` to `[u8; 32]` so that we could use it as a key in `BTreeMap`.

Probably a better way to address this (and other similar instances) is to implement `Ord` for `Digest` but this change needs to happen in Winterfell.

Reply:
Maybe we should make an issue in Winterfell for this change, since it came up during memoization as well and will probably come up again in the future. Wdyt?

Reply:
Agreed! Though since we'll be switching to RPO in the future (for v0.4), we might just make this change in our `crypto` repo ([here](https://github.com/0xPolygonMiden/crypto/blob/main/src/hash/rpo/digest.rs#L12)). I'll make an issue there.